packages/core/integration-tests/test/scope-hoisting.js

```diff
@@ -5532,4 +5533,82 @@ describe('scope hoisting', function () {
     let output = await run(b);
     assert.strictEqual(output, 'foo');
   });
+
+  it('produce the same bundle hash regardless of transformation order', async function () {
+    let testDir = path.join(
+      __dirname,
+      'integration/scope-hoisting/es6/non-deterministic-bundle-hashes',
+    );
+
+    const waitHandler = (fileToDelay, fileToWaitFor) => {
+      const waitMap = new Map();
+
+      function wait(filePath) {
+        if (waitMap.has(filePath)) {
+          return Promise.resolve();
+        }
+        return new Promise(resolve => {
+          waitMap.set(filePath, resolve);
+        });
+      }
+      // a set of filepaths that have been read
+      function seen(filePath) {
+        // check map of things we're waiting for to resolved promises
+        let promisesToResolve = waitMap.get(filePath);
+        if (promisesToResolve) {
+          // if we find any, we call it
+          promisesToResolve();
+        }
+        waitMap.set(filePath, null);
+      }
+
+      return {
+        get(target, prop) {
+          let original = Reflect.get(...arguments);
+          if (prop === 'readFile') {
+            return async function (...args) {
+              if (args[0].includes(fileToDelay)) {
+                await wait(fileToWaitFor);
+              }
+              let result = await original.apply(this, args);
+              seen(path.basename(args[0]));
+              return result;
+            };
+          }
+          return original;
+        },
+      };
+    };
+
+    let slowFooFS = new Proxy(overlayFS, waitHandler('foo.js', 'bar.js'));
+
+    let b = await bundle(path.join(testDir, 'index.html'), {
+      inputFS: slowFooFS,
+      outputFS: slowFooFS,
+      shouldDisableCache: true,
+      workerFarm: workerFarmZeroConcurrent,
+    });
+
+    let bundleHashDelayFoo = b
+      .getBundles()
+      .find(b => b.filePath.endsWith('.js') && b.filePath.includes('index'))
+      .filePath.split('.')[1];
+
+    let slowBarFS = new Proxy(overlayFS, waitHandler('bar.js', 'foo.js'));
+
+    let b2 = await bundle(path.join(testDir, 'index.html'), {
+      inputFS: slowBarFS,
+      outputFS: slowBarFS,
+      shouldDisableCache: true,
+      workerFarm: workerFarmZeroConcurrent,
+    });
+
+    let bundleHashDelayBar = b2
+      .getBundles()
+      .find(b => b.filePath.endsWith('.js') && b.filePath.includes('index'))
+      .filePath.split('.')[1];
+
+    assert.strictEqual(bundleHashDelayFoo, bundleHashDelayBar);
+    await workerFarmZeroConcurrent.end();
```

Comment:
Having a global `workerFarmZeroConcurrent` and then ending it in one test is weird, so it would be better to keep it as a local variable here. And also add a `try {..} finally { end(); }` around it in case the test fails.
But I'm also out of ideas on why CI is failing even with the end call

Reply:
thanks! is this what you mean? 
https://github.com/parcel-bundler/parcel/pull/8212/files#diff-ee074b5a5bbe809758dc3be2de9a23752857e0022abe2ccb350a3cbe28ad4018R5585-R5590